server/agents/casparcg-scanner/casparcg-scanner-agent.mjs

```diff
@@ -0,0 +1,127 @@
+import fetch from 'node-fetch'
+import path from 'path/win32'
+export { CasparCGScannerAgent }
+
+const paths = {
+	SEARCH: 'media',
+	STILLS: 'media/thumbnail/'
+}
+
+const REQUESTS = {
+	CLIP_SEARCH: {
+		path: paths.SEARCH
+	}
+}
+
+/** Agent for quering a CasparCG Scanner media scanner */
+class CasparCGScannerAgent {
+	/**
+	 * Create an agent.
+	 *
+	 * @param {string} host - Address to the Quantel server to query
+	 */
+
+	constructor(host, basePath) {
+		this.host = host
+		this.basePath = basePath
+	}
+
+	/**
+	 * Search for clips matching the given criteria.
+	 *
+	 * Special note on the created criteria:
+	 * Solr date search syntax used. Example for everything created within the last 2 days:
+	 * [NOW-2DAY/DAY TO NOW]
+	 *
+	 * @param {object} criteria - query criteria
+	 * @param {string} criteria.title - clip title criteria. * is allowed as a wildcard
+	 * @param {string} criteria.created - scope the search to clips created in a specific period
+	 *
+	 * @returns {Promise} - a promise containing the search results
+	 */
+
+	searchClip(criteria) {
+		const { path } = REQUESTS.CLIP_SEARCH
+
+		const url = new URL(this.host)
+		url.pathname = url.pathname + path
+
+		return fetch(url.href)
```

Comment:
This `searchClips` function is returning a promise, so it could be made async and this promise chain could be flattened out